decidim/decidim
# Fix HTML validation regarding SVG images

#### :tophat: What? Why?
In #7663 I added the `role="presentation"` to the SVG images which was suggested at:
https://github.com/dequelabs/axe-core/issues/1462

And this is not even the only resource if you search online.

The problem is that this breaks HTML validation, as you can see here:
https://github.com/validator/validator/issues/981

I will not revert #7663 because it still applies some other important changes but the take from all this is:

- All `<svg>` images need `role="img"` to pass HTML validation
- When the `<svg>` has a `role="img"` it needs to have an alternative text through `aria-label` to pass accessibility validation
- As I've understood, the `<svg>`'s also need the `<title>` element (which needs to match the `aria-label`)
- From other accessibility audits, we've learned that if the SVG title is the default title (icon name in English), it will be an accessibility violation when the page is shown in some other language than English. Therefore, if the `title` and `aria-label` are undefined for the icon, we now force the `aria-hidden="true"` attribute on the SVG.

This PR makes both accessibility evaluation tools and HTML validation tools happy.

**SIDENOTE:** We need some automated testing to the CI pipeline to catch all these, so we don't have to change it back and forth and fix over and over again. I'll come back to this once we get all these accessibility fixes merged, I have an idea how to implement it.

#### :pushpin: Related Issues
- Related to #7663, #5942, #6246, #6124, #5684

#### Testing
1. Go to https://validator.w3.org/#validate_by_input
2. Paste the following content in the direct input field:
```
<!doctype html>
<html lang="en">
<head><title>Test</title></head>
<body>
<svg role="presentation"><title>test</title><use href="/path/to/icons.svg#icon-test"></use></svg>
</body>
</html>
```
3. See the validation error

#### :clipboard: Checklist
